tests/integration_custom_event_check.tftest.hcl

```diff
@@ -0,0 +1,669 @@
+variables {
+  resource_name = "opslevel_integration_custom_event_check"
+
+  # required fields
+  name = "TF Test custom_event_check Integration"
+  type = "deploy"
+
+  # optional fields
+}
+
+run "resource_integration_custom_event_check_create_api_doc_type" {
+
+  variables {
+    type = "apiDoc"
+  }
+
+  module {
+    source = "./opslevel_modules/modules/integration/custom_event_check"
+  }
+
+  assert {
+    condition = opslevel_integration_custom_event_check.this.type == var.type
+    error_message = format(
+      "expected '%v' but got '%v'",
+      var.type,
+      opslevel_integration_custom_event_check.this.type,
+    )
+  }
+}
+
+run "resource_integration_custom_event_check_update_api_doc_type" {
+
+  variables {
+    name = "apiDoc Updated"
+    type = "apiDoc"
+  }
+
+  module {
+    source = "./opslevel_modules/modules/integration/custom_event_check"
+  }
+
+  assert {
+    condition = opslevel_integration_custom_event_check.this.type == var.type
+    error_message = format(
+      "expected '%v' but got '%v'",
+      var.type,
+      opslevel_integration_custom_event_check.this.type,
+    )
+  }
+}
+
+run "resource_integration_custom_event_check_create_aqua_security_type" {
+
+  variables {
+    type = "aquaSecurity"
+  }
+
+  module {
+    source = "./opslevel_modules/modules/integration/custom_event_check"
+  }
+
+  assert {
+    condition = opslevel_integration_custom_event_check.this.type == var.type
+    error_message = format(
+      "expected '%v' but got '%v'",
+      var.type,
+      opslevel_integration_custom_event_check.this.type,
+    )
+  }
+}
+
+run "resource_integration_custom_event_check_create_argocd_type" {
+
+  variables {
+    type = "argocd"
+  }
+
+  module {
+    source = "./opslevel_modules/modules/integration/custom_event_check"
+  }
+
+  assert {
+    condition = opslevel_integration_custom_event_check.this.type == var.type
+    error_message = format(
+      "expected '%v' but got '%v'",
+      var.type,
+      opslevel_integration_custom_event_check.this.type,
+    )
+  }
+}
+
+run "resource_integration_custom_event_check_create_aws_ecr_type" {
+
+  variables {
+    type = "awsEcr"
+  }
+
+  module {
+    source = "./opslevel_modules/modules/integration/custom_event_check"
+  }
+
+  assert {
+    condition = opslevel_integration_custom_event_check.this.type == var.type
+    error_message = format(
+      "expected '%v' but got '%v'",
+      var.type,
+      opslevel_integration_custom_event_check.this.type,
+    )
+  }
+}
```

Comment:
Should we add a CLI delete call in here somewhere to ensure it automatically re-creates them?

Reply:
Changing the type enacts a force replacement so we're good here. It's kind of a one off